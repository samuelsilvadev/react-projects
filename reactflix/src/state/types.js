export const ADD_VIDEO = 'video:ADD_VIDEO';

export const OPEN_REGISTER_VIDEO_FORM = 'ui:OPEN_REGISTER_VIDEO_FORM';
export const CLOSE_REGISTER_VIDEO_FORM = 'ui:CLOSE_REGISTER_VIDEO_FORM';

