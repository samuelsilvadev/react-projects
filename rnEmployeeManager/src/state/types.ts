export interface Action {
	type: string;
	payload?: any;
}

export interface User {}

export interface AUTH {
	email: string;
	password: string;
	user: User;
	isLoading: boolean;
	error: any;
}

export interface EMPLOYEE_FORM {
	name: string;
	phone: string;
	shift: string;
	isLoading: boolean;
	error: any;
}

export interface STATE {
	auth: AUTH;
	employeeForm: EMPLOYEE_FORM;
	employees: Object;
}

export const EMAIL_CHANGED = '@FORM/EMAIL_CHANGED';
export const PASSWORD_CHANGED = '@FORM/PASSWORD_CHANGED';

export const LOGIN_USER_LOADING = '@FORM/LOGIN_USER_LOADING';
export const LOGIN_USER_ERROR = '@FORM/LOGIN_USER_ERROR';
export const LOGIN_USER_SUCCESS = '@FORM/LOGIN_USER_SUCCESS';

export const EMPLOYEE_FIELD_UPDATE = '@EMPLOYEE/EMPLOYEE_FIELD_UPDATE';
export const EMPLOYEE_RESET = '@EMPLOYEE/EMPLOYEE_RESET';
export const EMPLOYEES_FETCH_SUCCESS = '@EMPLOYEE/EMPLOYEES_FETCH_SUCCESS';
