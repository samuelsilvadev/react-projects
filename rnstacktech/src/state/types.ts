export type Action = {
	type: String;
	payload: any;
};

export const SELECT_LIBRARY = 'SELECT_LIBRARY';
