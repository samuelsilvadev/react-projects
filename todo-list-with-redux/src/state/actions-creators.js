import { v4 } from 'uuid';
import { ADD_TODO, TOGGLE_TODO } from './types';

export const addTodo = (text) => {
	return ({
		type: ADD_TODO,
		payload: {
			id: v4(),
			text,
		},
	});
};

export const toggleTodo = (id) => {
	return ({
		type: TOGGLE_TODO,
		payload: {
			id,
		},
	});
};
