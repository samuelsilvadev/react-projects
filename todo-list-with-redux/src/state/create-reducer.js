const createReducer = (initialState, actionsMap = {}) => {
	return (state = initialState, action = {}) => {
		if (actionsMap.hasOwnProperty(action.type)) {
			return actionsMap[action.type](state, action);
		}

		return state;
	}
};

export default createReducer;
