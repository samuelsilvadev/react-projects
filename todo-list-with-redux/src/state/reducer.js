import { ADD_TODO, TOGGLE_TODO } from './types';

const INITIAL_STATE = [];

const todos = (state = INITIAL_STATE, action  = {}) => {
	if (action.type === ADD_TODO) {
		return state.concat({
			id: action.payload.id,
			text: action.payload.text,
			completed: false,
		});
	}

	if (action.type === TOGGLE_TODO) {
		return state.map((todo) => {
			if (todo.id === action.payload.id) {
				return {
					...todo,
					completed: !todo.completed,
				}
			}

			return todo;
		});
	}

	return state;
};

export default todos;
