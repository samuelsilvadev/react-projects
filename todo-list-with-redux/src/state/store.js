import { createStore } from 'redux';
import rootReducer from './reducers/root';

export const store = createStore(rootReducer);
