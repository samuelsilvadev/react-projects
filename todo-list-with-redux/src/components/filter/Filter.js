import React from 'react';
import { connect } from 'react-redux';

import { SHOW_ACTIVE, SHOW_ALL, SHOW_COMPLETED } from './../../state/types';
import { setVisibilityFilter } from './../../state/actions-creators';

import './Filter.css';

export const FILTERS =  [
	{ label: 'All', action: SHOW_ALL },
	{ label: 'Completed', action: SHOW_COMPLETED },
	{ label: 'Todo', action: SHOW_ACTIVE },
];

export const Filter = ({ activeFilter, updateFilter }) => {
	const _handleClick = (filter) => () => {
		updateFilter(filter);
	};
	
	const _renderFilterItem = (filter) => {
		const isDisabled = activeFilter === filter.action;

		return (
			<button
				key={ filter.action }
				data-enzyme-id={ `filter-button-${filter.label}` }
				className='btn'
				disabled={ isDisabled }
				onClick={ !isDisabled ? _handleClick(filter.action) : undefined }>{ filter.label }</button>
		);
	};

	return (
		<section data-enzyme-id="filter">
			<h2 data-enzyme-id="filter-title">Filter by:</h2>
			{ FILTERS.map(_renderFilterItem) }
		</section>
	);
};

const mapStateToProps = (state) => ({
	activeFilter: state.visibilityFilters,
});

const mapDispatchToProps = (dispatch) => ({
	updateFilter: (filter) => dispatch(setVisibilityFilter(filter)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Filter);
