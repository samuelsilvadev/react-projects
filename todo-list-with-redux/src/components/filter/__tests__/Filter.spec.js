import React from 'react';
import { shallow } from 'enzyme';

import { SHOW_ALL } from './../../../state/types';

import { Filter, FILTERS } from './../Filter';

describe('<Filter />', () => {
	it('should render correctly', () => {
		const tree = shallow(<Filter />);
		const $filter = tree.find('[data-enzyme-id="filter"]');
		const $title = tree.find('[data-enzyme-id="filter-title"]');

		FILTERS.forEach((filter) => {
			const $button = tree.find(`[data-enzyme-id="filter-button-${filter.label}"]`);

			expect($button).toBeDefined();
			expect($button.text()).toBe(filter.label);
		});

		expect($filter).toBeDefined();
		expect($title.text()).toBe('Filter by:');
		expect(tree).toMatchSnapshot();
	});

	it('should call `updateFilter` when click on `button`', () => {
		const updateFilter = jest.fn();

		const tree = shallow(<Filter updateFilter={ updateFilter } />);
		const $filterButtonCompleted = tree.find('[data-enzyme-id="filter-button-Completed"]');

		$filterButtonCompleted.simulate('click');

		expect(updateFilter).toHaveBeenCalledTimes(1);
	});

	it('should not call `updateFilter` if `button` is disabled', () => {
		const updateFilter = jest.fn();

		const tree = shallow(<Filter updateFilter={ updateFilter } activeFilter={ SHOW_ALL } />);
		const $filterButtonAll = tree.find('[data-enzyme-id="filter-button-All"]');

		$filterButtonAll.simulate('click');

		expect($filterButtonAll.is('[disabled]')).toBe(true);
		expect(updateFilter).not.toHaveBeenCalledTimes(1);
	});
});
