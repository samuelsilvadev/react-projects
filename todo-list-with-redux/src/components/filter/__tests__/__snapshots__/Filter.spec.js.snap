// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Filter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          data-enzyme-id="filter-title"
        >
          Filter by:
        </h2>,
        Array [
          <button
            className="btn"
            data-enzyme-id="filter-button-All"
            disabled={false}
            onClick={[Function]}
          >
            All
          </button>,
          <button
            className="btn"
            data-enzyme-id="filter-button-Completed"
            disabled={false}
            onClick={[Function]}
          >
            Completed
          </button>,
          <button
            className="btn"
            data-enzyme-id="filter-button-Todo"
            disabled={false}
            onClick={[Function]}
          >
            Todo
          </button>,
        ],
      ],
      "data-enzyme-id": "filter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Filter by:",
          "data-enzyme-id": "filter-title",
        },
        "ref": null,
        "rendered": "Filter by:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "visibilityFilter:SHOW_ALL",
        "nodeType": "host",
        "props": Object {
          "children": "All",
          "className": "btn",
          "data-enzyme-id": "filter-button-All",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "All",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "visibilityFilter:SHOW_COMPLETED",
        "nodeType": "host",
        "props": Object {
          "children": "Completed",
          "className": "btn",
          "data-enzyme-id": "filter-button-Completed",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Completed",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "visibilityFilter:SHOW_ACTIVE",
        "nodeType": "host",
        "props": Object {
          "children": "Todo",
          "className": "btn",
          "data-enzyme-id": "filter-button-Todo",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Todo",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            data-enzyme-id="filter-title"
          >
            Filter by:
          </h2>,
          Array [
            <button
              className="btn"
              data-enzyme-id="filter-button-All"
              disabled={false}
              onClick={[Function]}
            >
              All
            </button>,
            <button
              className="btn"
              data-enzyme-id="filter-button-Completed"
              disabled={false}
              onClick={[Function]}
            >
              Completed
            </button>,
            <button
              className="btn"
              data-enzyme-id="filter-button-Todo"
              disabled={false}
              onClick={[Function]}
            >
              Todo
            </button>,
          ],
        ],
        "data-enzyme-id": "filter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Filter by:",
            "data-enzyme-id": "filter-title",
          },
          "ref": null,
          "rendered": "Filter by:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "visibilityFilter:SHOW_ALL",
          "nodeType": "host",
          "props": Object {
            "children": "All",
            "className": "btn",
            "data-enzyme-id": "filter-button-All",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "All",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "visibilityFilter:SHOW_COMPLETED",
          "nodeType": "host",
          "props": Object {
            "children": "Completed",
            "className": "btn",
            "data-enzyme-id": "filter-button-Completed",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Completed",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "visibilityFilter:SHOW_ACTIVE",
          "nodeType": "host",
          "props": Object {
            "children": "Todo",
            "className": "btn",
            "data-enzyme-id": "filter-button-Todo",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Todo",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
